xbmc/addons/binary-addons/BinaryAddonManager.h

```diff
@@ -0,0 +1,113 @@
+#pragma once
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "addons/AddonManager.h"
+#include "threads/CriticalSection.h"
+
+#include <map>
+
+namespace ADDON
+{
+
+  class IAddonInstanceHandler;
+
+  class CAddonDll;
+  typedef std::shared_ptr<CAddonDll> AddonDllPtr;
+
+  class CBinaryAddonBase;
+  typedef std::shared_ptr<CBinaryAddonBase> BinaryAddonBasePtr;
+  typedef std::vector<BinaryAddonBasePtr> BinaryAddonBaseList;
+
+  class CBinaryAddonManager
+  {
+  public:
+    CBinaryAddonManager() = default;
+    ~CBinaryAddonManager();
+
+    bool Init();
+
+    /*!
+     * @brief Checks system about given type to know related add-on's are
+     * installed.
+     *
+     * @param[in] type Add-on type to check installed
+     * @return true if given type is installed
+     */
+    bool HasInstalledAddons(const TYPE &type);
+
+    /*!
+     * @brief Checks system about given type to know related add-on's are
+     * installed and also minimum one enabled.
+     *
+     * @param[in] type Add-on type to check enabled
+     * @return true if given type is enabled
+     */
+    bool HasEnabledAddons(const TYPE &type);
+
+    /*!
+     * @brief Get a list of add-on's with info's for the on system available
+     * ones.
+     *
+     * @param[out] addonInfos list where finded addon information becomes stored
+     * @param[in] enabledOnly If true are only enabled ones given back,
+     *                        if false all on system available. Default is true.
+     * @param[in] type        The requested type, with "ADDON_UNKNOWN"
+     *                        are all add-on types given back who match the case
+     *                        with value before.
+     *                        If a type id becomes added are only add-ons
+     *                        returned who match them. Default is for all types.
+     */
+    void GetAddonInfos(BinaryAddonBaseList& addonInfos, bool enabledOnly, const TYPE &type);
```

Comment:
Has tried with several ways, has produced always errors, with the try to write on addonInfos.